### PR TITLE
Document global credentials auth config for WND

### LINKED ISSUE #768: ACL support for WND
## What Needs to be Documented?

https://github.com/owncloud/windows_network_drive/pull/234

## Why Should This Change Be Made? (Optional)

New content planned for the next WND release. It will require documentation because there are several things to notice about the implementation.

## What Type Of Content Change Is This? (Optional)
- [x] New Content Addition
- [ ] Old Content Deprecation
- [ ] Existing Content Simplification
- [ ] Bug Fix to Existing Content

## Which Manual Does This Relate To? (Optional)
- [x] Admin Manual
- [ ] Developer Manual
- [ ] User Manual
- [ ] Android
- [ ] iOS
- [ ] Branded Clients
- [ ] Desktop Client
- [ ] Other

@settermjd there is some information in the PR, but there could be more information that we might need to add. This should be for the next WND (not sure when). I can provide additional information about the implementation, so feel free to ping me.
Also pinging to @felixboehm and @PVince81 for them to be aware

### PR DESCRIPTION
**Pull request by settermjd**
This relates to owncloud/windows_network_drive#242, and fixes #768.

### CODE REVIEW

File: modules/admin_manual/pages/enterprise/external_storage/windows-network-drive_configuration.adoc [hunk 1]
```diff
@@ -174,6 +175,42 @@ for login credentials:
...
+  For example, the ownCloud user "Alice" with the password `mysecret` (or even without a password) uses `Alice` as their username to obtain access to the WND storage. 
++
+NOTE: For LDAP, the user id is usually `1223-cbdf-cacc-1234...`, unless the configuration in {ldap-app-url}[the user_ldap app] is changed.
+
+* *The password is fetched from a key inside a `config.php.` file*.
+  Even if the account doesn't have a password (`oAuth`, `OpenIDConnect`, or another authentication mechanism), the connection with the storage swill use the password from the configuration file.
++
+NOTE: The password is the same for all users.
+
+Once this authentication mechanism is selected, a new field is available in the mount configuration, sin order to set the config key we want to use.
```
Comment (jvillafanez) [inline modules/admin_manual/pages/enterprise/external_storage/windows-network-drive_configuration.adoc]: typo: "sin order"

File: modules/admin_manual/pages/enterprise/external_storage/windows-network-drive_configuration.adoc [hunk 2]
```diff
@@ -174,6 +175,42 @@ for login credentials:
...
 * Global credentials
 
+==== Global Credentials
+
+To authenticate with global credentials:
+
+* *The username is the same as the user id of the ownCloud session.* 
+  For example, the ownCloud user "Alice" with the password `mysecret` (or even without a password) uses `Alice` as their username to obtain access to the WND storage. 
++
+NOTE: For LDAP, the user id is usually `1223-cbdf-cacc-1234...`, unless the configuration in {ldap-app-url}[the user_ldap app] is changed.
```
Comment (jvillafanez) [inline modules/admin_manual/pages/enterprise/external_storage/windows-network-drive_configuration.adoc]: is it worthy to point what they need to change here? I mean, I guess this point to the user_ldap configuration, but the admin can start touching things randomly.

**Comment by settermjd** [comment]
Thanks for the feedback, @jvillafanez. Will implement it.

File: modules/admin_manual/pages/enterprise/external_storage/windows-network-drive_configuration.adoc (hunk 2 above)
Comment (settermjd) [reply modules/admin_manual/pages/enterprise/external_storage/windows-network-drive_configuration.adoc]: Well said. I removed that line.

File: modules/admin_manual/pages/enterprise/external_storage/windows-network-drive_configuration.adoc [hunk 3]
```diff
@@ -174,6 +175,42 @@ for login credentials:
 * Log-in credentials, save in database
 * Global credentials
 
+==== Global Credentials
```
[the remark this answers is not in the document]
Comment (settermjd) [reply modules/admin_manual/pages/enterprise/external_storage/windows-network-drive_configuration.adoc]: Fair enough. I changed the name and called it Config-based credentials. If you do know what it's called, specifically, let me know and I'll change the name, unless you're keen to do that.

File: modules/admin_manual/pages/enterprise/external_storage/windows-network-drive_configuration.adoc (hunk 1 above)
Comment (settermjd) [reply modules/admin_manual/pages/enterprise/external_storage/windows-network-drive_configuration.adoc]: Fixed.

**Comment by settermjd** [comment]
@jvillafanez, can you take one further look over the PR, pls?

File: modules/admin_manual/pages/enterprise/external_storage/windows-network-drive_configuration.adoc [hunk 4]
```diff
@@ -165,14 +166,48 @@ NOTE: When you create a new mountpoint using Login credentials, you must log out
...
 
 * Username and password
 * Log-in credentials, save in session
 * Log-in credentials, save in database
 * Global credentials
+* Config-based credentials
+
+==== Config-based Credentials
+
+Config-based credentials provides a universal account for accessing an ownCloud instance.
```
Comment (jvillafanez) [inline modules/admin_manual/pages/enterprise/external_storage/windows-network-drive_configuration.adoc]: "universal account" sounds to me that it's only one account. Each user will have a different account although the password for all of them will be the same.

File: modules/admin_manual/pages/enterprise/external_storage/windows-network-drive_configuration.adoc [hunk 5]
```diff
@@ -165,14 +166,48 @@ NOTE: When you create a new mountpoint using Login credentials, you must log out
...
+The username is the same as the user id of the ownCloud session. 
+For example, the ownCloud user "Alice" with the password `mysecret` (or even without a password) uses `Alice` as their username to obtain access to the WND storage. 
+
+The password is fetched from a key inside a `config.php.` file.
+Even if the account doesn't have a password (`oAuth`, `OpenIDConnect`, or another authentication mechanism), the connection with the storage swill use the password from the configuration file.
```
Comment (jvillafanez) [inline modules/admin_manual/pages/enterprise/external_storage/windows-network-drive_configuration.adoc]: typo: "swill"

**Comment by jvillafanez** [comment]
Should we include a use case where this type of credentials could be useful? Otherwise it could be misused. I mean, having all the accounts the same password isn't a good idea on first sight.

File: modules/admin_manual/pages/enterprise/external_storage/windows-network-drive_configuration.adoc (hunk 4 above)
Comment (settermjd) [reply modules/admin_manual/pages/enterprise/external_storage/windows-network-drive_configuration.adoc]: Well said. I'll find another name for it.

File: modules/admin_manual/pages/enterprise/external_storage/windows-network-drive_configuration.adoc (hunk 5 above)
Comment (settermjd) [reply modules/admin_manual/pages/enterprise/external_storage/windows-network-drive_configuration.adoc]: I had thought I'd caught them all.

**Comment by jvillafanez** [comment]
@settermjd no changes shown :confused: , I can't review.

**Pull request closed without merging**